### PR TITLE
[ET-792] Add tickets link to each event

### PR DESCRIPTION
**Pull request by cliffordp**
Started from https://github.com/moderntribe/event-tickets/pull/1624, which got auto-closed by my renaming the branch.

[ET-792]

1min demo: https://share.getcloudapp.com/Wnub4rkR

[ET-792]: https://moderntribe.atlassian.net/browse/ET-792

### CODE REVIEW

File: src/views/shortcodes/my-attendance-list.php [hunk 1]
```diff
@@ -6,22 +6,41 @@
...
  * @var array $event_ids
  */
+
+$view = Tribe__Tickets__Tickets_View::instance();
 ?>
 
 <ul class="tribe-tickets my-attendance-list">
-	<?php foreach ( $event_ids as $id ): ?>
+	<?php
+	foreach ( $event_ids as $id ):
```
Comment (tr1b0t) [inline src/views/shortcodes/my-attendance-list.php:22]: * :no_entry_sign: Overriding WordPress globals is prohibited. Found assignment to $id
* :no_entry_sign: Expected 1 space(s) after closing parenthesis; found 0
* :no_entry_sign: You must have a space after a closing parenthesis
* :no_entry_sign: Space between opening control structure and T_COLON is required

<sub>hash: 82c030ec7db1eb360fe5349fc83be2d0</sub>

File: src/views/shortcodes/my-attendance-list.php [hunk 2]
```diff
@@ -6,22 +6,41 @@
...
+		$is_event = function_exists( 'tribe_is_event' ) ? tribe_is_event( $id ) : false;
```
Comment (tr1b0t) [inline src/views/shortcodes/my-attendance-list.php:23]: :warning: Equals sign not aligned with surrounding assignments; expected 15 spaces but found 1 space

<sub>hash: 8dd56ba9346ec5030a5371f29e6d0a8d</sub>

File: src/views/shortcodes/my-attendance-list.php [hunk 3]
```diff
@@ -6,22 +6,41 @@
...
 				<?php endif; ?>
 			</a>
+			<?php
+			if ( ! empty( $direct_link_my_tickets ) ) :
+				?>
+				<span class="event-post-tickets-separator">&mdash;</span>
+				<a class="event-post-direct-tickets-link" href="<?php echo esc_url( $direct_link_my_tickets ); ?>">
+					<?php esc_html_e( 'View Tickets', 'event-tickets' ); ?>
+				</a>
+			<?php
```
Comment (tr1b0t) [inline src/views/shortcodes/my-attendance-list.php:41]: :no_entry_sign: Line indented incorrectly; expected at least 4 tabs, found 3

<sub>hash: 8bd1cd620e345b542862e7d5734056ff</sub>

File: src/views/shortcodes/my-attendance-list.php (hunk 1 above)
Comment (sc0ttkclark) [reply src/views/shortcodes/my-attendance-list.php:22]:
```suggestion
	foreach ( $event_ids as $id ) :
```

File: src/views/shortcodes/my-attendance-list.php [hunk 4]
```diff
@@ -6,22 +6,41 @@
...
+
+$view = Tribe__Tickets__Tickets_View::instance();
 ?>
 
 <ul class="tribe-tickets my-attendance-list">
-	<?php foreach ( $event_ids as $id ): ?>
+	<?php
+	foreach ( $event_ids as $id ):
+		$is_event = function_exists( 'tribe_is_event' ) ? tribe_is_event( $id ) : false;
+		$direct_link_my_tickets = $view->get_tickets_page_url( $id, $is_event );
```
Comment (sc0ttkclark) [inline src/views/shortcodes/my-attendance-list.php:24]:
```suggestion
		$is_event               = function_exists( 'tribe_is_event' ) ? tribe_is_event( $id ) : false;
		$direct_link_my_tickets = $view->get_tickets_page_url( $id, $is_event );
```

File: src/views/shortcodes/my-attendance-list.php (hunk 3 above)
Comment (sc0ttkclark) [reply src/views/shortcodes/my-attendance-list.php:41]:
```suggestion
				<?php
```

File: src/views/shortcodes/my-attendance-list.php [hunk 5]
```diff
@@ -6,22 +6,41 @@
...
+		?>
 		<?php $start_date = tribe_get_start_date( $id ); ?>
 		<li class="event-<?php echo esc_attr( $id ) ?>">
-			<a href="<?php echo esc_url( get_permalink( $id ) ); ?>" target="_blank">
+			<a class="event-post-link" href="<?php echo esc_url( get_permalink( $id ) ); ?>">
```
Comment (sc0ttkclark) [inline src/views/shortcodes/my-attendance-list.php:28]:
```suggestion
			<a href="<?php echo esc_url( get_permalink( $id ) ); ?>" class="event-post-link">
```

File: src/views/shortcodes/my-attendance-list.php [hunk 6]
```diff
@@ -6,22 +6,41 @@
...
 				<?php echo get_the_title( $id ); ?>
 				<?php if ( $start_date ): ?>
 					<span class="datetime">(<?php echo $start_date; ?>)</span>
 				<?php endif; ?>
 			</a>
+			<?php
+			if ( ! empty( $direct_link_my_tickets ) ) :
+				?>
+				<span class="event-post-tickets-separator">&mdash;</span>
+				<a class="event-post-direct-tickets-link" href="<?php echo esc_url( $direct_link_my_tickets ); ?>">
```
Comment (sc0ttkclark) [inline src/views/shortcodes/my-attendance-list.php:38]:
```suggestion
				<a href="<?php echo esc_url( $direct_link_my_tickets ); ?>" class="event-post-direct-tickets-link">
```

File: src/views/shortcodes/my-attendance-list.php [hunk 7]
```diff
@@ -6,22 +6,41 @@
...
  * @var array $event_ids
  */
+
+$view = Tribe__Tickets__Tickets_View::instance();
 ?>
 
 <ul class="tribe-tickets my-attendance-list">
-	<?php foreach ( $event_ids as $id ): ?>
+	<?php
+	foreach ( $event_ids as $id ) :
```
Comment (tr1b0t) [inline src/views/shortcodes/my-attendance-list.php:22]: :no_entry_sign: Overriding WordPress globals is prohibited. Found assignment to $id

<sub>hash: 94442278494cd8b968f1fbfaf76794f8</sub>

**Comment by cliffordp** [comment]
@sc0ttkclark please approve with renamed variable and artifact in initial comment.

File: src/views/shortcodes/my-attendance-list.php [hunk 8]
```diff
@@ -6,22 +6,42 @@
...
-		<li class="event-<?php echo esc_attr( $id ) ?>">
-			<a href="<?php echo esc_url( get_permalink( $id ) ); ?>" target="_blank">
-				<?php echo get_the_title( $id ); ?>
+	<?php
+	foreach ( $event_ids as $event_id ) :
+		$is_event               = function_exists( 'tribe_is_event' ) ? tribe_is_event( $event_id ) : false;
+		$direct_link_my_tickets = $view->get_tickets_page_url( $event_id, $is_event );
+		?>
+		<?php $start_date = tribe_get_start_date( $event_id ); ?>
+		<li class="event-<?php echo esc_attr( $event_id ) ?>">
```
Comment (tr1b0t) [inline src/views/shortcodes/my-attendance-list.php:28]: :no_entry_sign: Inline PHP statement must end with a semicolon

<sub>hash: 82f99327e22129da2cabf31270355177</sub>

File: src/views/shortcodes/my-attendance-list.php [hunk 9]
```diff
@@ -6,22 +6,42 @@
...
+			<a href="<?php echo esc_url( get_permalink( $event_id ) ); ?>" class="event-post-link">
+				<?php echo get_the_title( $event_id ); ?>
```
Comment (tr1b0t) [inline src/views/shortcodes/my-attendance-list.php:30]: :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found 'get_the_title'.

<sub>hash: 83dfefc66d1105b831e5f3605ea4d0cc</sub>

Comment (cliffordp) [reply src/views/shortcodes/my-attendance-list.php:30]: `the_title()` can't be used here for whatever reason it outputs differently, but it also doesn't get escaped from `get_the_title()` so I didn't modify this

File: src/views/shortcodes/my-attendance-list.php (hunk 8 above)
Comment (cliffordp) [reply src/views/shortcodes/my-attendance-list.php]: fixed

**Pull request merged**